actionpack/lib/action_dispatch/railtie.rb

```diff
@@ -39,20 +39,23 @@ class Railtie < Rails::Railtie # :nodoc:
     config.eager_load_namespaces << ActionDispatch
 
     initializer "action_dispatch.configure" do |app|
-      ActionDispatch::Http::URL.tld_length = app.config.action_dispatch.tld_length
-      ActionDispatch::Request.ignore_accept_header = app.config.action_dispatch.ignore_accept_header
-      ActionDispatch::Request::Utils.perform_deep_munge = app.config.action_dispatch.perform_deep_munge
-      ActionDispatch::Response.default_charset = app.config.action_dispatch.default_charset || app.config.encoding
-      ActionDispatch::Response.default_headers = app.config.action_dispatch.default_headers
-      ActionDispatch::Response.return_only_media_type_on_content_type = app.config.action_dispatch.return_only_media_type_on_content_type
-
-      ActionDispatch::ExceptionWrapper.rescue_responses.merge!(config.action_dispatch.rescue_responses)
-      ActionDispatch::ExceptionWrapper.rescue_templates.merge!(config.action_dispatch.rescue_templates)
-
-      config.action_dispatch.always_write_cookie = Rails.env.development? if config.action_dispatch.always_write_cookie.nil?
-      ActionDispatch::Cookies::CookieJar.always_write_cookie = config.action_dispatch.always_write_cookie
-
-      ActionDispatch.test_app = app
+      config = self.config
+      ActiveSupport.on_load(:action_dispatch) do
```

Comment:
Hmm. Is this approach really works expected? 
`ActionDispatch` is already loaded in this context. So `ActiveSupport.on_load(:action_dispatch)` executes immediately I think.

Reply:
Yeah, this would need to be a load hook for `ActionDispatch::Response` in order to be effective. Unfortunately, `ActionDispatch::Response` is eager loaded at boot time too:

https://github.com/rails/rails/blob/9680984cf8ce608294e8ee161d40eb87d7426a5a/actionpack/lib/action_controller.rb#L5
https://github.com/rails/rails/blob/9680984cf8ce608294e8ee161d40eb87d7426a5a/actionpack/lib/action_controller/metal/live.rb#L3

Maybe we can fix `ActionDispatch::Response` to autoload properly and add a hook.

Reply:
I opened https://github.com/rails/rails/pull/37228 to fix the autoload problem with `ActionDispatch::Response`.